presto-cli/src/test/java/io/prestosql/cli/TestInsecureQueryRunner.java

```diff
@@ -0,0 +1,190 @@
+        return sslContext;
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void teardown() throws Exception
+    {
+        server.close();
+    }
+
+    @Test
+    public void testInsecureConnection() throws Exception
+    {
+        server.enqueue(new MockResponse()
+                .setResponseCode(307)
+                .addHeader(LOCATION, server.url("/v1/statement"))
+                .addHeader(SET_COOKIE, "a=apple"));
+        server.enqueue(new MockResponse()
+                .addHeader(CONTENT_TYPE, "application/json")
+                .setBody(createResults()));
+        server.enqueue(new MockResponse()
+                .addHeader(CONTENT_TYPE, "application/json")
+                .setBody(createResults()));
+
+        ClientSession clientSession = new ClientSession(
+                server.url("/").uri(),
+                "user",
+                "source",
+                Optional.empty(),
+                ImmutableSet.of(),
+                "clientInfo",
+                "catalog",
+                "schema",
+                "path",
+                ZoneId.of("America/Los_Angeles"),
+                Locale.ENGLISH,
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                null,
+                new Duration(2, MINUTES));
+
+        QueryRunner queryRunner = new QueryRunner(
+                clientSession,
+                true,
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                false,
+                true);
+
+        try (Query query = queryRunner.startQuery("first query will introduce a cookie")) {
+            query.renderOutput(nullPrintStream(), nullPrintStream(), CSV, false, false);
+        }
+        try (Query query = queryRunner.startQuery("second query should carry the cookie")) {
+            query.renderOutput(nullPrintStream(), nullPrintStream(), CSV, false, false);
+        }
+        try {
+            assertEquals(server.takeRequest().getHeader("Cookie"), null);
+            assertEquals(server.takeRequest().getHeader("Cookie"), "a=apple");
+            assertEquals(server.takeRequest().getHeader("Cookie"), "a=apple");
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String createResults()
```

Comment:
This looks the same as the method in `TestQueryRunner`. If so, let's change that one to be a static, package-private method which takes `server` as a parameter, so we can reuse it here.